pydrad/parse/util.py

```diff
@@ -0,0 +1,228 @@
+            pop_arrays[j][i, :] = line[1:]  # Skip first entry, it is the atomic number
+    columns = []
+    for z in Z:
+        el_name = plasmapy.particles.element_name(z)
+        columns += [f'{el_name}_{i+1}' for i in range(z+1)]
+    data = np.hstack([p for p in pop_arrays])
+    return astropy.table.QTable(data=data, names=columns)
+
+
+def read_trm_file(filename):
+    """
+    Parse ``.trm`` files with hydrodynamic equation terms as a function of position.
+
+    The files come in sets of 5 rows with variable number of columns:
+        -- Loop coordinate (1 column), and at each position:
+        -- Terms of mass equation (2 columns)
+        -- Terms of momentum equation (6 columns)
+        -- Terms of electron energy equation (11 columns)
+        -- Terms of hydrogen energy equation (11 columns)
+    """
+    units = {
+        'mass': 'g cm-3 s-1',
+        'momentum': 'dyne cm-3 s-1',
+        'electron': 'erg cm-3 s-1',
+        'hydrogen': 'erg cm-3 s-1',
+    }
+    columns = {
+        'mass': [
+            'mass_drhobydt',
+            'mass_advection',
+        ],
+        'momentum': [
+            'momentum_drho_vbydt',
+            'momentum_advection',
+            'momentum_pressure_gradient',
+            'momentum_gravity',
+            'momentum_viscous_stress',
+            'momentum_numerical_viscosity',
+        ],
+        'electron': [
+            'electron_dTEKEbydt',
+            'electron_enthalpy',
+            'electron_conduction',
+            'electron_gravity',
+            'electron_collisions',
+            'electron_heating',
+            'electron_radiative_loss',
+            'electron_electric_field',
+            'electron_viscous_stress',
+            'electron_numerical_viscosity',
+            'electron_ionization'
+        ],
+        'hydrogen': [
+            'hydrogen_dTEKEbydt',
+            'hydrogen_enthalpy',
+            'hydrogen_conduction',
+            'hydrogen_gravity',
+            'hydrogen_collisions',
+            'hydrogen_heating',
+            'hydrogen_radiative_loss',
+            'hydrogen_electric_field',
+            'hydrogen_viscous_stress',
+            'hydrogen_numerical_viscosity',
+            'hydrogen_ionization',
+        ]
+    }
+    tables = []
+    for i,(k,v) in enumerate(columns.items()):
+        tables.append(
+            astropy.table.QTable.from_pandas(
+                read_csv(
+                    filename,
+                    sep='\t',
+                    header=None,
+                    names=v,
+                    skiprows=lambda x: x % 5 != (i+1)
+                ),
+                units={c: units[k] for c in v}
+            )
+        )
+    return astropy.table.hstack(tables)
+
+
+def read_hstate_file(filename):
+    """
+    Parse the ``.hstate`` files containing the hydrogen level populations as a function of position
+    """
+    columns = [f'level_population_hydrogen_{i}' for i in range(1,7)]
```

Comment:
`level_population_hydrogen_3` is not really understandable at first glance.  Level 6 is also _not_ the 6th level, it's the ionized fraction.

Is there a better naming scheme we could use?

Reply:
Also, perhaps adding a "neutral fraction" property, which is just the sum of levels 1 through 5, might be helpful.  It's sometimes helpful to consider ions vs neutrals in the chromosphere.

Reply:
I'm open to any better naming scheme! I think I probably named these somewhat arbitrarily as I didn't know what else to call them.

Also 👍 on a derived neutral fraction property. I would suggest that be its own PR.

Reply:
Perhaps just simplifying `hydrogen_level_i` for the first five, and `ionized_fraction` for the last.  I'm not sure if there's a much better name without abbreviations, but alternatively maybe `HI_level_i` and `HII_fraction`.  Not sure.